### PR TITLE
Reading Logs

### PR DESCRIPTION
**Pull request by carson-ricca**

### CODE REVIEW

File: reading_logs_retrieval.py [hunk 1]
```diff
@@ -15,4 +17,15 @@
     except IndexError:
         canvas_course_id = get_default_course_id()
 
-    canvas_reading_logs_retrieval(EXPORT_DIR, encoder, canvas_course_id)
+    canvas_reading_logs_retrieval(EXPORT_DIR, encoder, canvas_course_id, ASSIGNMENT_IDS[0])
```
Comment (carson-ricca) [inline reading_logs_retrieval.py:20]: I understand that this looks bad lol. But every time I iterated assignments in a loop to get every attachment for every submission, after completing the first assignment the CanvasApi would timeout every time, and not allow any subsequent downloads. Running it individually for each assignment was the only way I could get it to download every assignment without API issues.

File: scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py [hunk 2]
```diff
@@ -1,46 +1,43 @@
...
-        output_path = setup_reading_logs_filepath(course_reading_logs_dir, 'reading_logs', assignment)
-        assignment_submissions = assignment.get_submissions()
-        download_reading_logs(assignment_submissions, output_path, encoder)
+    output_path = setup_reading_logs_filepath(course_reading_logs_dir, 'reading_logs')
+    assignment_submissions = assignment.get_submissions()
+    download_reading_logs(assignment_submissions, output_path, encoder)
 
 
-def setup_reading_logs_filepath(parent_dir: str, sub_dir: str, assignment) -> str:
+def setup_reading_logs_filepath(parent_dir: str, sub_dir: str) -> str:
```
Comment (carson-ricca) [inline scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py:30]: The util function in another PR will not be able to replace this due to the file extension that is part of the util function.

File: util/file.py [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
 
 
-def mkdir_if_not_exists(path: str) -> str:
+def mkdir_if_not_exists(path: str, multiple_directories=False) -> str:
     if not os.path.exists(path):
-        os.mkdir(path)
+        if not multiple_directories:
+            os.mkdir(path)
+        else:
+            os.makedirs(path)
```
Comment (carson-ricca) [inline util/file.py:9]: Required to be able to make multiple directories at once.

File: reading_logs_retrieval.py (hunk 1 above)
Comment (nononovia) [reply reading_logs_retrieval.py:20]: Damn, that's unfortunate...

File: scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py (hunk 2 above)
Comment (nononovia) [reply scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py:30]: Sounds good, thank you for pointing that out.

File: scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py [hunk 4]
```diff
@@ -1,46 +1,43 @@
...
     """
     Get the correct filepath for a reading logs download
     :param parent_dir: the parent directory path
     :param sub_dir: the name of the sub directory for this object type's submissions to be saved
     :param assignment: the assignment we are retrieving the reading logs for
```
Comment (nononovia) [inline scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py:35]: the parameter assignment is removed, the docstring for assignment should probably be removed too.

**Review by Rick-Feng-u — APPROVED** [review]

**Review by nononovia — CHANGES_REQUESTED** [review]
Looks good, this seems like a rough task, and wow you came up with great solutions, thankss

File: scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py (hunk 4 above)
Comment (carson-ricca) [reply scripts/canvas_reading_logs_retrieval/canvas_reading_logs_retrieval.py:35]: Good catch!

**Review by nononovia — APPROVED** [review]
lgtm

**Pull request merged**